include/ensmallen_bits/age_moea/agemoea.hpp

```diff
@@ -0,0 +1,489 @@
+  void FastNonDominatedSort(std::vector<std::vector<size_t>>& fronts,
+                            std::vector<size_t>& ranks,
+                            std::vector<arma::Col<typename MatType::elem_type> >& calculatedObjectives);
+
+  /**
+   * Operator to check if one candidate Pareto-dominates the other.
+   *
+   * A candidate is said to dominate the other if it is at least as good as the
+   * other candidate for all the objectives and there exists at least one
+   * objective for which it is strictly better than the other candidate.
+   *
+   * @tparam MatType Type of matrix to optimize.
+   * @param calculatedObjectives The previously calculated objectives.
+   * @param candidateP The candidate being compared from the elite population.
+   * @param candidateQ The candidate being compared against.
+   * @return true if candidateP Pareto dominates candidateQ, otherwise, false.
+   */
+  template<typename MatType>
+  bool Dominates(std::vector<arma::Col<typename MatType::elem_type>>& calculatedObjectives,
+                 size_t candidateP,
+                 size_t candidateQ);
+
+ /**
+  * Assigns Survival Score metric for sorting.
+  *
+  * @param front The previously generated Pareto fronts.
+  * @param idealPoint The ideal point of teh first front.
+  * @param calculatedObjectives The previously calculated objectives.
+  * @param survivalScore The Survival Score vector to be updated for each individual in the population.
+  * @param normalize The normlization vector of the fronts.
+  * @param dimension The dimension of the first front.
+  * @param fNum teh current front index.
+  */
+  template <typename MatType>
+  void SurvivalScoreAssignment(const std::vector<size_t>& front,
+    const arma::Col<typename MatType::elem_type>& idealPoint,
+    std::vector<arma::Col<typename MatType::elem_type>>& calculatedObjectives,
+    std::vector<typename MatType::elem_type>& survivalScore,
+    arma::Col<typename MatType::elem_type>& normalize,
+    double& dimension,
+    size_t fNum);
+
+  /**
+   * The operator used in the AGE-MOEA survival score based sorting.
+   *
+   * If a candidate has a lower rank then it is preferred.
+   * Otherwise, if the ranks are equal then the candidate with the larger
+   * Survival Score is preferred.
+   *
+   * @param idxP The index of the first cadidate from the elite population being
+   *     sorted.
+   * @param idxQ The index of the second cadidate from the elite population
+   *     being sorted.
+   * @param ranks The previously calculated ranks.
+   * @param survivalScore The Survival score for each individual in
+   *	the population.
+   * @return true if the first candidate is preferred, otherwise, false.
+   */
+  template<typename MatType>
+  bool SurvivalScoreOperator(size_t idxP,
+                             size_t idxQ,
+                             const std::vector<size_t>& ranks,
+                             const std::vector<typename MatType::elem_type>& survivalScore);
```

Comment:
```suggestion
  bool SurvivalScoreOperator(
      size_t idxP,
      size_t idxQ,
      const std::vector<size_t>& ranks,
      const std::vector<typename MatType::elem_type>& survivalScore);
```
Same here, just to keep lines from being too long.